editor.planx.uk/src/@planx/components/List/Public/Context.tsx

```diff
@@ -24,50 +37,109 @@ export const ListProvider: React.FC<ListProviderProps> = ({
   children,
   schema,
 }) => {
-  const [activeIndex, setActiveIndex] = useState<number | undefined>(0);
-  const [userData, setUserData] = useState<UserData>(
-    schema.min === 0 ? [] : [generateNewItem(schema)],
-  );
+  const [activeIndex, setActiveIndex] = useState<number>(0);
+
+  const [addItemError, setAddItemError] = useState<boolean>(false);
+  const [unsavedItemError, setUnsavedItemError] = useState<boolean>(false);
+  const [minError, setMinError] = useState<boolean>(false);
+  const [maxError, setMaxError] = useState<boolean>(false);
```

Comment:
Setting these two via the validation schema proved surprisingly tricky as `schema.max` is optional. I'm handling these manually here for now.